src/Core/Style.js

```diff
@@ -549,7 +554,12 @@ class Style {
 
         this.icon = {};
         defineStyleProperty(this, 'icon', 'source', params.icon.source);
-        defineStyleProperty(this, 'icon', 'key', params.icon.key);
+        if (params.icon.key) {
+            console.warn('Deprecated: use icon.id instead');
```

Comment:
Can you add what is deprecated ? e.g. : `icon.key is deprecated, use icon.id instead`

Reply:
done